client/src/services/DonuService.ts

```diff
@@ -12,19 +13,85 @@ const callDonu = (request: Donu.Request): Promise<Donu.Response> => {
   }
 };
 
+export const getDonuModelSource = async (model: string, type: Donu.Type): Promise<Donu.ModelGraph> => {
+  const request: Donu.Request = {
+    command: Donu.RequestCommand.GET_MODEL_SOURCE,
+    definition: {
+      type,
+      source: {
+        model,
+      },
+    },
+  };
+  const response = await callDonu(request);
+  if (response.status === Donu.ResponseStatus.success) {
+    const modelSource = response.result.source;
+    return JSON.parse(modelSource);
+  } else {
+    console.error('[DONU Service] — fetchDonuModelSource', response);
+  }
+};
+
+const populateDonuModelWithGrometSource = async (model: Donu.ModelDefinition): Promise<void> => {
```

Comment:
Why this needs to be a separate `const` it looks small enough to be part of step 3 `map()`

Reply:
I'll mark this as resolve